### PR TITLE
Relatively stable focus-based & hotkey-toggled visibility.

### PR DESCRIPTION
**Pull request by sqt**
Slight modification to how the hotkey toggle worked in order to combine approaches and use a single handler (now both called via Dispatcher).  Excessive calls to browserWindow.Show() were bad news for the focus-checking side of things, so there are status flags and explicit checks for when the state needs to change before calling Show().  Hide() doesn't seem to have the same issue?

### CODE REVIEW

**Comment by quisquous** [comment]
This seems to work in most cases, but if I do the following:
(1) focus the game
(2) hit ctrl+esc to bring up the start menu
(3) click on something open other than the game/ACT on the task bar (?)
(4) click on the game window

This seems to make it disappear when it shouldn't.  If I then alt+tab to ACT, I can get it back? It's a little odd, though.

File: ACTPlugin.cs [hunk 1]
```diff
@@ -48,14 +54,52 @@ public void InitPlugin(TabPage pluginScreenSpace, Label pluginStatusText)
...
-                    browserWindow.Show();
-            };
+            {
+                overlayEnabled = !overlayEnabled;
+                currentDispatcher.Invoke(visCheck);
+            };
+            System.Windows.Automation.AutomationFocusChangedEventHandler afceh = (sender, o) =>
+            {
+                System.Windows.Automation.AutomationElement focusedElement = sender as System.Windows.Automation.AutomationElement;
+                if (focusedElement != null)
```
Comment (quisquous) [inline ACTPlugin.cs]: Code review nitpicking me suggests early return instead of giant indented block?

File: ACTPlugin.cs [hunk 2]
```diff
@@ -48,14 +54,52 @@ public void InitPlugin(TabPage pluginScreenSpace, Label pluginStatusText)
...
+                {
+                    int processId = focusedElement.Current.ProcessId;
+                    using (System.Diagnostics.Process process = System.Diagnostics.Process.GetProcessById(processId))
```
Comment (quisquous) [inline ACTPlugin.cs]: Also, do you really need this using an indented block if you only use |process| once?

File: ACTPlugin.cs (hunk 1 above)
Comment (sqt) [inline ACTPlugin.cs]: Single return point habits kicking in.  I have no strong preference here.

File: ACTPlugin.cs (hunk 2 above)
Comment (sqt) [inline ACTPlugin.cs]: Not knowing what "using" was for, it looked like other pieces of code were using it to save space on null checks, but after looking it up it looks like it's secretly wrapping this with a try/catch block as well, which I don't really care about.  I'll pull this out into another check / early return.

**Pull request closed without merging**